dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -84,6 +133,30 @@ public static void main(String[] arguments) throws IOException, ClassNotFoundExc
     System.out.println(stringBuilder.toString());
   }
 
+  /**
+   * Finds jar file paths for the dependencies of the Maven coordinate.
+   *
+   * @param coordinate Maven coordinate of an artifact to check its dependencies
+   * @return list of absolute paths to jar files
+   * @throws RepositoryException when there is a problem in retrieving jar files
+   */
+  static List<Path> coordinateToJarPaths(String coordinate) throws RepositoryException {
+    DefaultArtifact artifact = new DefaultArtifact(coordinate);
+    DependencyGraph transitiveDependencies =
+        DependencyGraphBuilder.getTransitiveDependencies(artifact);
+    List<DependencyPath> dependencyPaths = transitiveDependencies.list();
```

Comment:
If there are multiple versions of an artifact on the classpath, will this process them both?

Reply:
Yes, the list would includes both in that case.

However, the order may not be what Maven uses (BFS)...

Reply:
This is precisely why I think we should not have our own app try to figure out the jars; we should get the literal classpath passed in from Maven. We want to check exactly what the user will see when they run their app or use their library.

Reply:
> Yes, the list would includes both in that case.

This was incorrect. It only list one for an artifact.

Reply:
For grpc-auth 1.15.1 , https://storage.googleapis.com/cloud-opensource-java-dashboard/dashboard/target/dashboard/io.grpc_grpc-auth_1.15.1.html

 DependencyGraphBuilder.getTransitiveDependencies gives:

```
/Users/suztomo/.m2/repository/io/grpc/grpc-auth/1.15.1/grpc-auth-1.15.1.jar, 
 /Users/suztomo/.m2/repository/io/grpc/grpc-core/1.15.1/grpc-core-1.15.1.jar,
 /Users/suztomo/.m2/repository/io/grpc/grpc-context/1.15.1/grpc-context-1.15.1.jar,
 /Users/suztomo/.m2/repository/com/google/code/gson/gson/2.7/gson-2.7.jar,
 /Users/suztomo/.m2/repository/com/google/guava/guava/20.0/guava-20.0.jar,
 /Users/suztomo/.m2/repository/com/google/errorprone/error_prone_annotations/2.2.0/error_prone_annotations-2.2.0.jar,
 /Users/suztomo/.m2/repository/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar,
 /Users/suztomo/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.17/animal-sniffer-annotations-1.17.jar,
 /Users/suztomo/.m2/repository/io/opencensus/opencensus-api/0.12.3/opencensus-api-0.12.3.jar,
 /Users/suztomo/.m2/repository/io/opencensus/opencensus-contrib-grpc-metrics/0.12.3/opencensus-contrib-grpc-metrics-0.12.3.jar,
 /Users/suztomo/.m2/repository/com/google/auth/google-auth-library-credentials/0.9.0/google-auth-library-credentials-0.9.0.jar]
```

Reply:
I see. It looks like at least for the cases we have coded for, it's correct, so maybe we can go with this for now. I'm still concerned about edge cases long term though, and I don't trust us to match Maven's class path in 100% of cases.

Reply:
@elharo do you have an opinion on this matter?

Reply:
I'm not sure what you mean by the "literal classpath passed in from Maven". We can handle multiple classpath's and if we can get maven to tell us which classpath it's using under what circumstances, then we can use that with this code. Off the top of my head, I don't know how to get Maven to tell us that.

Reply:
We started with Maven plugins. However we rapidly ran into several related problems:

1. The plugins are not designed to be invoked as a library. You have to spawn a process.
2. The plugins only operate on a source code project with a pom.xml, not an artifact in the repo. 
3. The output of the plugins isn't trivially parseable. It's designed to be read, not processed.